src/loaders/workspace_uploader/workspace_uploader.py

```diff
@@ -0,0 +1,571 @@
+    assembly_name: str,
+) -> str:
+    """Upload an assembly file to workspace."""
+    success, attempts, max_attempts = False, 0, 3
+    while attempts < max_attempts and not success:
+        try:
+            time.sleep(attempts)
+            assembly_ref = conf.asu.save_assembly_from_fasta2(
+                {
+                    "file": {"path": file_path},
+                    "workspace_name": workspace_name,
+                    "assembly_name": assembly_name,
+                }
+            )
+            success = True
+        except Exception as e:
+            print(f"Error:\n{e}\nfrom attempt {attempts + 1}.\nTrying to rerun.")
+            attempts += 1
+
+    if not success:
+        raise ValueError(
+            f"Upload Failed for {file_path} after {max_attempts} attempts!"
+        )
+
+    upa = assembly_ref["upa"].replace("/", "_")
+    return upa
+
+
+def _read_upload_status_yaml_file(
+        upload_env_key: str,
+        workspace_id: int,
+        assembly_dir: str,
+        assembly_name: str, 
+) -> Tuple[dict[str, dict[int, list[str]]], bool]:
+    """
+    Get metadata and upload status of an assembly from the uploaded.yaml file.
+    """
+
+    uploaded = False
+    if upload_env_key not in loader_common_names.KB_ENV:
+        raise ValueError(f"Currently only support these {loader_common_names.KB_ENV} envs for upload")
+
+    file_path = _get_yaml_file_path(assembly_dir)
+
+    with open(file_path, "r") as file:
+        data = yaml.safe_load(file)
+
+    if not data:
+        data = {upload_env_key: dict()}
+
+    if workspace_id not in data[upload_env_key]:
+        data[upload_env_key][workspace_id] = dict()
+
+    assembly_dict = data[upload_env_key][workspace_id]
+    if assembly_dict and assembly_dict["file_name"] == assembly_name:
+        uploaded = True
+    return data, uploaded
+
+
+def _update_upload_status_yaml_file(
+        upload_env_key: str,
+        workspace_id: int,
+        upa: str,
+        assembly_dir: str,
+        assembly_name: str,
+) -> None:
+    """
+    Update the uploaded.yaml file in target genome_dir with newly uploaded assembly names and upa info.
+    """
+    data, uploaded = _read_upload_status_yaml_file(upload_env_key, workspace_id, assembly_dir, assembly_name)
+
+    if uploaded:
+        raise ValueError(f"Assembly {assembly_name} already exists in workspace {workspace_id}")
+
+    data[upload_env_key][workspace_id] = {"file_name": assembly_name, "upa": upa}
+
+    file_path = _get_yaml_file_path(assembly_dir)
+    with open(file_path, "w") as file:
+        yaml.dump(data, file)
+
+
+def _fetch_assemblies_to_upload(
+        upload_env_key: str,
+        workspace_id: int,
+        collection_source_dir: str,
+        upload_file_ext: list[str],
+) -> Tuple[int, dict[str, str]]:
+    """
+    Help fetch assemblies to upload.
+    """
+    if upload_file_ext != UPLOAD_FILE_EXT:
+        raise ValueError(f"Currently only support assembly files to upload")
```

Comment:
We should allow any file extensions the user wants unless there's a good reason to prevent that

Reply:
: